pkg/appendable/io.go

```diff
@@ -254,11 +372,29 @@ func (f *IndexFile) Serialize(w io.Writer) error {
 			keys[i] = key
 			i++
 		}
+		var isCSVHandler bool
+		switch f.data.(type) {
+		case CSVHandler:
+			isCSVHandler = true
+		case JSONLHandler:
+			isCSVHandler = false
+		default:
+			panic("unknown handler type")
+		}
+
 		sort.Slice(keys, func(i, j int) bool {
 			at, bt := keys[i], keys[j]
-			if atr, btr := fieldRank(at), fieldRank(bt); atr != btr {
-				return atr < btr
+
+			if isCSVHandler {
+				if atr, btr := fieldRankCsvField(at), fieldRank(bt); atr != btr {
+					return atr < btr
+				}
+			} else {
+				if atr, btr := fieldRank(at), fieldRank(bt); atr != btr {
+					return atr < btr
+				}
 			}
```

Comment:
Instead of setting the boolean above it would be better to switch on the data handler and rank independently. I think it will be too hard to try to maintain a unified ranking.

```suggestion
		switch f.data.(type) {
		case CSVHandler:
			if atr, btr := fieldRankCsvField(at), fieldRankCsvField(bt); atr != btr {
				return atr < btr
			}
			// TODO: compare at, bt knowing that they are the same type.
		case JSONLHandler:
			if atr, btr := fieldRank(at), fieldRank(bt); atr != btr {
				return atr < btr
			}
			switch at.(type) {
			case nil:
				return false
			case bool:
				return !at.(bool) && bt.(bool)
			case int, int8, int16, int32, int64, float32, float64:
				return at.(float64) < bt.(float64)
			case string:
				return strings.Compare(at.(string), bt.(string)) < 0
			default:
				panic("unknown type")
			}
		default:
			panic("unknown handler")
		}
```

This ranking is probably better to put on `DataHandler` but that can be done in a separate PR.

Reply:
Yeah that was lazy on my part.